src/Umbraco.Core/Runtime/CoreRuntime.cs

```diff
@@ -224,6 +224,8 @@ public virtual void Compose(IServiceCollection services)
             // register persistence mappers - required by database factory so needs to be done here
             // means the only place the collection can be modified is in a runtime - afterwards it
             // has been frozen and it is too late
+#error Why does 
```

Comment:
For some reason, CollectionMapperBase is responsible for registering itself. Better to instantiate MapperCollectionBuilder and pass it IServiceCollection to register stuff while configuring itself? It can be registered as a singleton after anyway?